pkg/tfgen/generate_schema.go

```diff
@@ -520,11 +530,11 @@ func (g *schemaGenerator) genDatasourceFunc(mod string, fun *resourceFunc) psche
 
 	// If there are argument and/or return types, emit them.
 	if fun.argst != nil {
-		_, t := g.genObjectType(mod, &schemaNestedType{typ: fun.argst, pyMapCase: true})
+		_, t := g.genObjectType(mod, &schemaNestedType{typ: fun.argst, pyMapCase: true}, false)
```

Comment:
Can datasources have "id"?  I thought these would be true.

Reply:
I just checked - data sources can have an Id - but they can sometimes be optional for lookups. I'll change data sources to remove top level ids that are computed only

Reply:
For data sources, I omitted "id" outputs only - not inputs as that is a legit param